app/controllers/administration/cash_registers_controller.rb

```diff
@@ -0,0 +1,125 @@
+class Administration::CashRegistersController < AdministrationController
+
+  before_action :get_qualifiers, only: [:new, :show, :edit, :update, :create]
+  before_action :get_locations, only: [:new, :show, :edit, :update, :create]
+
+  before_action :check_kassa_account, only: [:create, :update]
+  before_action :check_pankkikortti_account, only: [:create, :update]
+  before_action :check_luottokortti_account, only: [:create, :update]
+  before_action :check_kateistilitys_account, only: [:create, :update]
+  before_action :check_kassaerotus_account, only: [:create, :update]
+  before_action :check_kateisotto_account, only: [:create, :update]
+
+  def index
+    @cash_registers = CashRegister
+    .search_like(search_params)
+    .order(order_params)
+  end
+
+  def show
+     render 'edit'
+  end
+
+  def edit
+  end
+
+  def new
+    @cash_register = CashRegister.new
+  end
+
+  def create
+    @cash_register = CashRegister.new(cash_register_params)
+
+    @cash_register.kustp = 0 if @cash_register.kustp.nil?
+
+    if @cash_register.save_by current_user
+      redirect_to cash_registers_path, notice: t("Kassalipas luotiin onnistuneesti")
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @cash_register.update_by(cash_register_params, current_user)
+      redirect_to cash_registers_path, notice: t("Kassalipas päivitettiin onnistuneesti")
+    else
+      render :edit
+    end
+  end
+
+private
+
+    def find_resource
+      @cash_register = CashRegister.find(params[:id])
+    end
+
+    def get_qualifiers
+      @qualifiers = Qualifier.all.order(:nimi)
+    end
+
+    def get_locations
+      @locations = Location.all.order(:nimi)
+    end
+
+    def check_kassa_account
```

Comment:
check_ metodeita käytetään ilmeisesti siihen, että sillä haetaan tilit dropdowneja varten? Tässä ehkä voisi harkita, että tarvitaanko näitä vai voisko näkymässä käyttää suoraan scopeja tjsp?